pkg/generator/generator.go

```diff
@@ -51,9 +52,16 @@ func New(log logrus.FieldLogger, s3Client s3wrapper.API, cfg Config, workDir str
 // GenerateInstallConfig creates install config and ignition files
 func (k *installGenerator) GenerateInstallConfig(ctx context.Context, cluster common.Cluster, cfg []byte, releaseImage string) error {
 	log := logutil.FromContext(ctx, k.log)
-	clusterWorkDir := filepath.Join(k.workDir, cluster.ID.String())
+	err := os.MkdirAll(k.workDir, 0o755)
+	if err != nil {
+		return err
+	}
+	clusterWorkDir, err := ioutil.TempDir(k.workDir, cluster.ID.String()+".")
+	if err != nil {
+		return err
+	}
 	installerCacheDir := filepath.Join(k.workDir, "installercache")
-	err := os.MkdirAll(clusterWorkDir, 0755)
+	err = os.MkdirAll(clusterWorkDir, 0755)
```

Comment:
i think it is not needed anymore

Reply:
@tsorya is right, you should have a `clusterWorkDir` by this time, `ioutil.TempDir(k.workDir, cluster.ID.String()+".") created it with the parent directory taken care of by `os.MkdirAll(k.workDir, 0o755)`.

Reply:
OK